typescript-sdk/src/grpc/fumarole.ts

```
@@ -0,0 +1,1790 @@
+// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
+// versions:
+//   protoc-gen-ts_proto  v1.181.2
+//   protoc               v6.30.0
+// source: fumarole.proto
+
+/* eslint-disable */
+import {
+  type CallOptions,
+  ChannelCredentials,
+  Client,
+  ClientDuplexStream,
+  type ClientOptions,
+  type ClientUnaryCall,
+  handleBidiStreamingCall,
+  type handleUnaryCall,
+  makeGenericClientConstructor,
+  Metadata,
+  type ServiceError,
+  type UntypedServiceImplementation,
+} from "@grpc/grpc-js";
+import Long from "long";
+import _m0 from "protobufjs/minimal";
+import {
+  CommitmentLevel,
+  commitmentLevelFromJSON,
+  commitmentLevelToJSON,
+  SubscribeRequestFilterAccounts,
+  SubscribeRequestFilterTransactions,
+  SubscribeUpdate,
+} from "./geyser";
+
+export const protobufPackage = "fumarole";
+
+export enum ConsumerGroupType {
+  STATIC = 0,
+  UNRECOGNIZED = -1,
+}
+
+export function consumerGroupTypeFromJSON(object: any): ConsumerGroupType {
+  switch (object) {
+    case 0:
+    case "STATIC":
+      return ConsumerGroupType.STATIC;
+    case -1:
+    case "UNRECOGNIZED":
+    default:
+      return ConsumerGroupType.UNRECOGNIZED;
+  }
+}
+
+export function consumerGroupTypeToJSON(object: ConsumerGroupType): string {
+  switch (object) {
+    case ConsumerGroupType.STATIC:
+      return "STATIC";
+    case ConsumerGroupType.UNRECOGNIZED:
+    default:
+      return "UNRECOGNIZED";
+  }
+}
+
+export enum InitialOffsetPolicy {
+  EARLIEST = 0,
+  LATEST = 1,
+  SLOT = 2,
+  UNRECOGNIZED = -1,
```

There should not be UNRECOGNIZED value in InitialOffsetPolicy from proto file.
We probablby want to fail fast if we get an expected value instead hidding it into "-1"